packages/web-components/fast-foundation/src/progress/progress.template.ts

```diff
@@ -21,24 +21,23 @@ export function progressTemplate(
             ${when(
                 x => typeof x.value === "number",
                 html<FASTProgress>`
-                    <div class="progress" part="progress" slot="determinate">
-                        <div
-                            class="determinate"
-                            part="determinate"
-                            style="width: ${x => x.percentComplete}%"
-                        ></div>
-                    </div>
+                    <span
+                        class="determinate"
+                        part="determinate"
+                        style="--percent-complete: ${x => x.percentComplete}"
+                    >
+                        ${options.determinateIndicator ||
+                        `<div class="indicator" part="indicator"></div>`}
+                    </span>
                 `
             )}
             ${when(
                 x => typeof x.value !== "number",
                 html<FASTProgress>`
-                    <div class="progress" part="progress" slot="indeterminate">
-                        <slot name="indeterminate">
-                            ${options.indeterminateIndicator1 || ""}
-                            ${options.indeterminateIndicator2 || ""}
-                        </slot>
-                    </div>
+                    <span class="indeterminate" part="indeterminate">
+                        ${options.indeterminateIndicator ||
+                        `<div class="indicator" part="indicator"></div>`}
+                    </span>
```

Comment:
I used this as a test for this pattern we have in a number of places. I might be missing something, but I don't understand the benefit of the slot / options combination. A slot makes sense for something where a given instance is going to have different content, like an icon in the 'start' slot on a button, but in the case of a progress bar (and others) I don't see the use case for swapping the indicator for a particular instance.

I see that the `options` is a convenient way to use the default template but swap out some of the content with something custom, which applies to _all_ instances of your progress bar.

I'd like to remove some of those slots if we don't have a good use case for swapping the content for individual instances.

Reply:
I don't think we should remove slots in favor of this pattern because slots are the web platform approach and this is simply syntactic sugar allowing the configuration of a *default*, it's not _the way_. Slots are the default and while I'm all for removing redundant approaches from a slot perspective, slots are the path forward and the default slotted content implementation is sugar for setting that globally for a template. It doesn't easily support the *continued* swapping out for one-off instances. We should not abandon the platform in favor of a sugar we've created for convenience on top of it.

cc: @EisenbergEffect in case he has additional thoughts here.

Reply:
I think we want to separate the slot vs. the slot's fallback content. The fallback content should only be provided (at this level in the stack) via an options property. We don't want to hard-code icons and such as defaults that folks may not want. But this is entirely different from slots. Slots are an architectural change to the component anatomy rather than just a content value difference.

In the above, I would just do this:

```
<span class="indeterminate" part="indeterminate">
   ${options.indeterminateIndicator}
</span>
```

The indicators can be extracted into exports that the CLI can use when generate the default options setup for a design system.

Reply:
@Eisenbergeffect are you proposing we go ahead with removing the slot? I'm not sure I'm in agreement there. I'd suggest that the slots need to stay and we should leverage options to deliver default/fallback content to them.